### DISCUSSION TITLE
3.0

### DISCUSSION THREAD

**Discussion by Rumperuu** [markcheret/footnotes#116] [Releases]
We are currently working towards the [2.7.1 release](https://github.com/markcheret/footnotes/milestone/3), which will contain a couple bugfixes, PHP compatibility checks, a reorganised settings menu and AMP compatibility, along with some stability improvements and a completed release workflow. After 2.7.1, we will switch to a less-frequent release schedule per #91.

The next major release after 2.7.1 will be [2.8.0](https://github.com/markcheret/footnotes/milestone/4), which will be a much more comprehensive review and refactor of the existing codebase. I've also moved #35 and #36 from the 2.7.1 milestone to the 2.8.0 milestone, as these will be part of that wider review and refactor.

This thread is for discussions about the 2.8.0 release. Some thoughts from me:

- I think 2.7.1 would be a sensible place to shift to using WordPress Versioning (per #92);
- it would also make sense to, as part of 2.8.0, update the pinned posts in the WP Support Forum and the `readme.txt` to reflect the new GitHub repo. and workflow; and
- see https://github.com/markcheret/footnotes/discussions/112#discussioncomment-623723

**Comment by markcheret** [comment]
- the readme.txt can be updated already and shipped with 2.7.3
- pinned post with github link is discouraged by forum mods. It's not our blog, it's a support forum. I will post on my website about it.

**Comment by Rumperuu** [comment]
Agreed in #112

**Comment by Rumperuu** [comment]
I've been caught up with other things for the last week or two, but here's a progress update on the [3.0 branch](https://github.com/markcheret/footnotes/tree/3.0):

- PHP 8.1 upgrade (and on-build PHP 7.4 downgrading) done
- docblocks updated and cleaned up
- Hungarian notation fully removed
- variable/property type declaration and return types implemented
- various other improvements

Next steps:

- implement automatic testing infrastructure
- split larger files like `class-parser.php` into smaller modules
- identify and remove additional unnecessary things (e.g. some of the conversion functions)

**Comment by lolzim** [comment]
Is AMP dead ?
https://developers.google.com/search/blog/2021/04/more-details-page-experience
